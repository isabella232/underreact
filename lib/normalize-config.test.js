'use strict';

const normalizeConfig = require('./normalize-config');
const path = require('path');
const getDefaultConfig = require('./default-underreact.config');

test('normalizes empty configuration', () => {
  const defaultConfig = getDefaultConfig({
    production: undefined,
    stats: undefined,
    port: undefined,
    configDir: path.join(process.cwd(), 'underreact.config.js')
  });
  expect(normalizeConfig({}, defaultConfig)).toMatchSnapshot();
});

test('normalizes fancy configuration', () => {
  const defaultConfig = getDefaultConfig({
    production: undefined,
    stats: undefined,
    port: undefined,
    configDir: path.join(process.cwd(), 'underreact.config.js')
  });
  expect(
    normalizeConfig(
      {
        polyfills: {
          promise: true,
          fetch: true
        },
        siteBasePath: 'fancy',
        publicAssetsPath: 'cacheable-things',
        stylesheets: [path.join(__dirname, './src/bg.css')],
        htmlSource: path.join(__dirname, './src/index.html'),
        babelPlugins: ['/babel-plugin-lodash'],
        webpackLoaders: [
          {
            test: /\.less$/,
            use: ['css-loader', 'less-loader']
          }
        ],
        webpackPlugins: [function foo() {}],
        environmentVariables: ['ENV_VAR'],
        webpackConfigTransform: config => {
          config.devtool = false;
          return config;
        }
      },
      defaultConfig
    )
  ).toMatchSnapshot();
});
