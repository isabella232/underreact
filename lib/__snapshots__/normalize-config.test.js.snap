// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizes empty configuration 1`] = `
Urc {
  "babelPlugins": Array [],
  "babelPresets": Array [],
  "devServerHistoryFallback": false,
  "environmentVariables": Array [],
  "htmlSource": "/Users/kushanjoshi/code/open/underreact/underreact.config.js/src/html.js",
  "jsEntry": "/Users/kushanjoshi/code/open/underreact/underreact.config.js/src/entry.js",
  "outputDirectory": "/Users/kushanjoshi/code/open/underreact/underreact.config.js/_underreact-site",
  "polyfills": Object {
    "fetch": false,
    "objectAssign": false,
    "promise": false,
  },
  "port": 8080,
  "postcssPlugins": Array [],
  "production": undefined,
  "publicAssetsPath": "/underreact-assets",
  "publicDirectory": "/Users/kushanjoshi/code/open/underreact/underreact.config.js/public",
  "rootDirectory": "/Users/kushanjoshi/code/open/underreact/underreact.config.js",
  "siteBasePath": "",
  "stats": undefined,
  "stylesheets": Array [],
  "vendorModules": Array [],
  "webpackConfigTransform": [Function],
  "webpackLoaders": Array [],
  "webpackPlugins": Array [],
}
`;

exports[`normalizes fancy configuration 1`] = `
Urc {
  "babelPlugins": Array [
    "/babel-plugin-lodash",
  ],
  "babelPresets": Array [],
  "devServerHistoryFallback": false,
  "environmentVariables": Array [
    "ENV_VAR",
  ],
  "htmlSource": "/Users/kushanjoshi/code/open/underreact/lib/src/index.html",
  "jsEntry": "/Users/kushanjoshi/code/open/underreact/underreact.config.js/src/entry.js",
  "outputDirectory": "/Users/kushanjoshi/code/open/underreact/underreact.config.js/_underreact-site",
  "polyfills": Object {
    "fetch": true,
    "objectAssign": false,
    "promise": true,
  },
  "port": 8080,
  "postcssPlugins": Array [],
  "production": undefined,
  "publicAssetsPath": "/cacheable-things",
  "publicDirectory": "/Users/kushanjoshi/code/open/underreact/underreact.config.js/public",
  "rootDirectory": "/Users/kushanjoshi/code/open/underreact/underreact.config.js",
  "siteBasePath": "/fancy",
  "stats": undefined,
  "stylesheets": Array [
    "/Users/kushanjoshi/code/open/underreact/lib/src/bg.css",
  ],
  "vendorModules": Array [],
  "webpackConfigTransform": [Function],
  "webpackLoaders": Array [
    Object {
      "test": /\\\\\\.less\\$/,
      "use": Array [
        "css-loader",
        "less-loader",
      ],
    },
  ],
  "webpackPlugins": Array [
    [Function],
  ],
}
`;
